nx-X11/programs/Xserver/hw/nxagent/Keystroke.c

```diff
@@ -440,17 +447,23 @@ int nxagentCheckSpecialKeystroke(XKeyEvent *X, enum HandleEventResult *result)
    * Do we need a cache ?
    */
 
-  sym = XKeycodeToKeysym(nxagentDisplay, X -> keycode, index);
+  int keysyms_per_keycode_return;
+  XlibKeySym *sym = XGetKeyboardMapping(nxagentDisplay,
+                                        X->keycode,
+                                        1,
+                                        &keysyms_per_keycode_return);
 
-  if (sym == XK_VoidSymbol || sym == NoSymbol)
+  if (*sym == XK_VoidSymbol || *sym == NoSymbol)
   {
+    free(sym);
     return 0;
   }
 
   #ifdef TEST
   fprintf(stderr, "nxagentCheckSpecialKeystroke: got code %x - state %x - sym %lx\n",
-              X -> keycode, X -> state, sym);
+              X -> keycode, X -> state, *sym);
   #endif
+  free(sym);
```

Comment:
The manpage says "To free the storage returned by XGetKeyboardMapping, use XFree." I think using free is the way to go nowadays but mentioning this just to be sure...